### PR TITLE
Prerendering/SSR fixes

### PR DESCRIPTION
**Pull request by patrickarlt**
This PR makes a bunch of fixes to allow Calcite Components to work with [Stencils prerendering system](https://stenciljs.com/docs/prerendering). I have a proof-of-concept of the prerendering working with Gatsby that I can share a little later.

### CODE REVIEW

File: conventions/README.md [hunk 1]
```diff
@@ -445,3 +446,32 @@ export class CalciteExample {
 ```
 
 This will create a unique id attribute like `id="calcite-example-51af-0941-54ae-22c14d441beb"` which should have a VERY low collision change since `guid()` generates IDs with `window.crypto.getRandomValues`. If a user supplies an `id` this will respect the users `id`.
+
+## Prerendering and SSR
+
+Stencil provide the capability to render web components on the server and seamlessly hydrate them on the client. This is handled by the `dist-hydrate-script` output target in `stencil.config.ts`.
+
+This generates a `hydrate` directory which expsoses `renderToString()` (for the server) and `hydarteDocument()` for the client.
```
Comment (paulcpederson) [inline conventions/README.md]: `hydarteDocument()` => `hydrateDocument()`

File: conventions/README.md [hunk 2]
```diff
@@ -445,3 +446,32 @@ export class CalciteExample {
...
+
+Since many of the same lifecycles methods are called on the client and server you may need to differentiate any code that relies on browser APIs like so:
```
Comment (paulcpederson) [inline conventions/README.md]: lifecycles => _lifecycle_

File: package.json [hunk 3]
```diff
@@ -31,7 +32,7 @@
     "url": "git+https://github.com/Esri/calcite-components.git"
   },
   "dependencies": {
-    "@a11y/focus-trap": "^1.0.2",
+    "@a11y/focus-trap": "git+https://github.com/patrickarlt/focus-trap.git#conditional-define-build",
```
Comment (paulcpederson) [inline package.json]: did you open a pr form this branch already?

**Comment by paulcpederson** [comment]
cool, this is a really neat feature, I had no idea this existed

**Comment by paulcpederson** [comment]
We should add a note about how to prerender these in the readme as well once it gets all sorted

File: package.json (hunk 3 above)
Comment (patrickarlt) [reply package.json]: Yup. This particular branch just has the built files in it already vs the one I PR'ed https://github.com/andreasbm/focus-trap/pull/1.

**Comment by paulcpederson** [comment]
Going to merge this in, nice work @patrickarlt :+1:

**Pull request merged**